README.md

```diff
@@ -1,6 +1,137 @@
 # stipulate
-A module extending the Fetch API with some useful default error handling and hooks.
+A module extending the Fetch API with some useful default error handling and data extraction.
 
 [![Build Status](https://travis-ci.org/yola/stipulate.svg?branch=master)](https://travis-ci.org/yola/stipulate)
+[![Coverage Status](https://coveralls.io/repos/github/yola/stipulate/badge.svg?branch=master)](https://coveralls.io/github/yola/stipulate?branch=master)
 
 Stipulate assumes the presence of a global `fetch`, in accordance with the [Fetch API spec](https://fetch.spec.whatwg.org/).
+
+## Usage
+
+```js
+import stipulate from 'stipulate';
+
+const options = {
+  credentials: 'same-origin',
+  headers: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  }
+};
+
+const responsePromise = stipulate('/some/endpoint', options);
+```
+
+If you need to extend options, you can use the `buildOptions` export.
+```js
+import { buildOptions } from 'stipulate';
+
+const opts = {
+  credentials: 'same-origin',
+  headers: {
+    'Accept': 'application/json',
+    'Content-Type': 'application/json'
+  }
+};
+
+const moreOpts = {
+  headers: {
+    'Cache-Control': 'no-cache'
+  }
+};
+
+const mergedOpts = buildOptions(opts, moreOpts);
+//  {
+//    credentials: 'same-origin',
+//    headers: {
+//      'Accept': 'application/json',
+//      'Cache-Control': 'no-cache',
+//      'Content-Type': 'application/json'
+//    }
+//  }
+```
+
+Null or empty string headers will not be written to merged options. If you pass `buildOptions` two
+options objects with the same header, one set to `null` or `""` and the other with a value, order matters.
+```js
+const nullAcceptHeaderOpts = {
+  headers: {
+    'Accept': null
+  }
+};
+
+const  = mergedOpts = buildOptions(nullAcceptHeaderOpts, opts);
+// buildOpts gives priority to first option set seen, so this would
+// produce:
+//  {
+//    credentials: 'same-origin',
+//    headers: {
+//      'Content-Type': 'application/json'
+//    }
+//  }
+
+buildOptions(opts, nullAcceptHeadersOpts);
+// this would produce:
+// {
+//   credentials: 'same-origin',
+//   headers: {
+//    'Accept': 'application/json',
+//    'Content-Type': 'application/json'
+//   }
+// }
+```
+
+`resolveUrl` is a tool for adding query parameters from an object to a url string.
+Duplicate keys will get resolved with priority going to values found on the query object.
+Like headers, you can override query params from the url by passing `null` or `''` values
+for those params in the query object.
+```js
+const query = {
+  foo: '',
+  zip: 'zap'
+};
+
+resolveUrl('http://some.domain/some/endpoint?foo=bar&fizz=buzz', query);
+// returns: "http://some.domain/some/endpoint?fizz=buzz&zip=zap"
+```
+
+### Errors
+
+By default, Stipulate throws responses with non 2XX status codes. There are two ways to modify this
```

Comment:
Since we're using promises, is it better to say "Stipulate rejects".

Otherwise it sounds like you'd need to wrap stipulate calls in a try/catch.